### PR TITLE
WP_Theme_JSON: Add nested indexed array schema sanitization capabilities.

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
- Add nested indexed array schema sanitization capabilities to the `WP_Theme_JSON` class.
- Add `settings.typography.fontFamilies` validation in `WP_Theme_JSON`
- Add unit tests.

## Why?
Currently, `WP_Theme_JSON` sanitization is not able to sanitize data contained on indexed arrays. So certain data from `theme.json`, for example, `settings.typography.fontFamilies` which is a JSON array, cannot be sanitized. Why? because when parsing the JSON data `WP_Theme_JSON` class translates JSON arrays into PHP indexed arrays and the it is not capable of sanitizing that kind of data.

---
Trac ticket: https://core.trac.wordpress.org/ticket/60360#

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by rovinasonaiz** [comment]
<a href='https://capmistoi-6281.weebly.com'>cache</a>
<a href='https://capmistoi-6282.weebly.com'>cache</a>
<a href='https://capmistoi-6283.weebly.com'>cache</a>
<a href='https://capmistoi-6284.weebly.com'>cache</a>
<a href='https://capmistoi-6285.weebly.com'>cache</a>
<a href='https://capmistoi-6286.weebly.com'>cache</a>
<a href='https://capmistoi-6287.weebly.com'>cache</a>
<a href='https://capmistoi-6288.weebly.com'>cache</a>
<a href='https://capmistoi-6289.weebly.com'>cache</a>
<a href='https://capmistoi-6290.weebly.com'>cache</a>
<a href='https://capmistoi-6291.weebly.com'>cache</a>
<a href='https://capmistoi-6292.weebly.com'>cache</a>
<a href='https://capmistoi-6293.weebly.com'>cache</a>
<a href='https://capmistoi-6294.weebly.com'>cache</a>
<a href='https://capmistoi-6295.weebly.com'>cache</a>
<a href='https://capmistoi-6296.weebly.com'>cache</a>
<a href='https://capmistoi-6297.weebly.com'>cache</a>
<a href='https://capmistoi-6298.weebly.com'>cache</a>
<a href='https://capmistoi-6299.weebly.com'>cache</a>
<a href='https://capmistoi-6300.weebly.com'>cache</a>

File: tests/phpunit/tests/theme/wpThemeJson.php [hunk 1]
```diff
@@ -4970,4 +4970,140 @@ public function test_get_stylesheet_custom_root_selector() {
...
+							),
+						),
+					),
+				),
+			),
+		);
+		$sanitized_theme_json = $theme_json->get_raw_data();
+		$this->assertSameSetsWithIndex( $expected_sanitized, $sanitized_theme_json, 'Sanitized theme.json does not match' );
+	}
+
```
Comment (mukeshpanchal27) [inline tests/phpunit/tests/theme/wpThemeJson.php]:
```suggestion
```

**Review by mukeshpanchal27 — COMMENTED** [review]
Fix PHPCS

**Comment by youknowriad** [comment]
Commit https://core.trac.wordpress.org/changeset/57496

**Pull request closed without merging**